src/script/lua_api/l_mainmenu.cpp

```diff
@@ -577,6 +577,13 @@ int ModApiMainMenu::l_get_favorites(lua_State *L)
 			lua_settable(L, top_lvl2);
 		}
 
+		if (servers[i]["ping"].asString().size()) {
```

Comment:
please store the string before condition and use !str.empty() and re-use the stored string into function instead of converting json value twice